Sources/LeafKit/LeafLexer.swift

```diff
@@ -132,123 +141,191 @@ struct LeafLexer {
+        if current.isValidOperator {
+            var op = Operator(rawValue: String(current) + String(src.peek()!))
+            if op == nil { op = Operator(rawValue: String(current)) } else { src.pop() }
+            if op != nil { return .parameter(.operator(op!)) }
+        }
+        
+        // Test for numerics next. This is not very intelligent but will read base2/8/10/16
+        // for Ints and base 10/16 for decimal through native Swift initialization
+        // Will not adequately decay to handle things like `0b0A` and recognize as invalid.
+        if current.canStartNumeric {
+            let next = src.peek()!
+            let peekRaw = String(current) + (src.peekWhile { $0.isValidInNumeric } ?? "")
+            var peekNum = peekRaw.replacingOccurrences(of: String(.underscore), with: "")
+            // 1 character string must be an integer
+            guard peekNum.count > 1 else { return .parameter(.constant(.int(Int(peekNum)!))) }
+                        
+            var testInt: Int?
+            var testDouble: Double?
+            var radix: Int?
+            
+            switch (peekNum.contains(.period), next, peekNum.count > 2) {
+                case (true, _, _) : testDouble = Double(peekNum)
+                case (false, .binaryNotation, true): radix = 2
+                case (false, .octalNotation, true): radix = 8
+                case (false, .hexNotation, true): radix = 16
+                default: testInt = Int(peekNum)
+            }
+            
+            if let radix = radix, radix != 10 {
+                let start = peekNum.startIndex
+                peekNum.removeSubrange(start ... peekNum.index(after: start))
+                testInt = Int(peekNum, radix: radix)
+            }
+            
+            if testInt != nil || testDouble != nil {
+                _ = src.popWhile { $0.isValidInNumeric }
+                if testInt != nil { return .parameter(.constant(.int(testInt!))) }
+                else { return .parameter(.constant(.double(testDouble!))) }
             }
-        case .body:
-            guard next == .colon else { fatalError("state should only be set to .body when a colon is in queue") }
-            src.pop()
-            state = .normal
-            return .tagBodyIndicator
+        }
+        
+        // At this point, just read anything that's parameter valid, but not an operator,
+        // Could be handled better and is probably way too aggressive.
+        let name = String(current) + (src.readWhile { $0.isValidInParameter && !$0.isValidOperator } ?? "")
+        
+        // If it's a keyword, return that
+        if let keyword = Keyword(rawValue: name) { return .parameter(.keyword(keyword)) }
+        // Assume anything that matches .isValidInTagName is a tag
+        // Parse can decay to a variable if necessary - checking for a paren
+        // is over-aggressive because a custom tag may not take parameters
```

Comment:
Just realized this comment is ambiguous since below it DOES check still check for a left paren before marking as a tag, but this is also ambiguous behavior when processing parameters. If you have a tag that can reasonable be used with no parameters, eg `#date` where no parameters would return default formatting and current date, then it's unclear in a parameter if `#if(date)` is referring to a variable or tag of that name. This is complicated since when NOT within a paramater, eg `#date` is clearly referring to a tag since a variable can't be scoped at the top level. This would be disambiguated by only allowing no-paren tags at the top level when they're body-expecting, eg: `#else:` (can't take parameters because it HAS to have a body) would be valid but `#date()` would be required for no-param use